bib/cmd/bootc-image-builder/main.go

```diff
@@ -313,16 +313,51 @@ func cmdBuild(cmd *cobra.Command, args []string) error {
 		if !slices.Contains(imgTypes, "ami") {
 			return fmt.Errorf("aws flags set for non-ami image type (type is set to %s)", strings.Join(imgTypes, ","))
 		}
-		// initialise the client to check if the env vars exist before building the image
+
+		// check as many permission prerequisites as possible before starting
 		client, err := awscloud.NewDefault(region)
 		if err != nil {
 			return err
 		}
 
-		fmt.Printf("Checking AWS permission by listing regions...\n")
-		if _, err := client.Regions(); err != nil {
-			return err
+		fmt.Printf("Checking AWS region access...\n")
+		var regions []string
```

Comment:
It's common in go to just avoid this var declaration as the call below will create `regions` and infer the type, so:
```go
-               var regions []string
-               regions, err = client.Regions();
+               regions, err := client.Regions()
```
should be fine here.

Reply:
but re-declares "err" :-/

Reply:
Not sure I follow - when I do the following:
```go
index 9f00131..5783e81 100644
--- a/bib/cmd/bootc-image-builder/main.go
+++ b/bib/cmd/bootc-image-builder/main.go
@@ -321,8 +321,7 @@ func cmdBuild(cmd *cobra.Command, args []string) error {
                }
 
                fmt.Printf("Checking AWS region access...\n")
-               var regions []string
-               regions, err = client.Regions()
+               regions, err := client.Regions()
                if err != nil {
                        return fmt.Errorf("retrieving AWS regions for '%s' failed: %w", region, err)
                }
```
it still compiles for me, go is fine with this because `regisions` is a new var so that is declared and err is just assigned, see https://go.dev/doc/effective_go#redeclaration and is quite common.

Or am I missing something?

Reply:
IntelliJ doesn't like this - but if it's an accepted re-declaration I'm happy to use this as it's nicer for sure!

Reply:
Out of curiosity, what does InteliJ say here? Is it a warning or an error? It's super common in go code to use this pattern of just redeclartions.

Reply:
Actually it's not even a warning its more a hint. The variable is just underlined as if it would be a typo :-/
I just checked and it should be a "helper" to notify me that I do redeclare. The underline just distracted me... I'll ignore from now on...